src/switch_core_io.c

```diff
@@ -1003,6 +1008,8 @@ SWITCH_DECLARE(switch_status_t) switch_core_session_read_frame(switch_core_sessi
 		}
 	}
 
+	switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(session), SWITCH_LOG_ERROR, "Read codec addr: %p, codec read mutex addr: %p\n", (void *)&session->read_codec, (void *)&session->codec_read_mutex);
```

Comment:
Do you think its best to validate NULL values then log the codec mutex are missing to lessen the noise? Since we are only interested if this mutex is NULL.

Reply:
Probably add two check, one is after the goto label `even_more_done` and in here. My worry is that the logging callback is process in a thread and when it crash it might not able to log it during the crash.

Reply:
I'm not sure if the mutex is NULL or some garbage pointer, but I can also add checks for NULL (a different log message perhaps?).

Reply:
right, i re-did my test, we can get segmentation fault in `__pthread_mutex_unlock_usercnt()` by either calling it with NULL or an invalid pointer, so there's no telling which it was